lib/vmpooler/api/v2.rb

```diff
@@ -146,6 +271,254 @@ def atomically_allocate_vms(payload)
         end
       end
 
+      def component_to_test(match, labels_string)
+        tracer.in_span("Vmpooler::API::V2.#{__method__}") do
+          return if labels_string.nil?
+  
+          labels_string_parts = labels_string.split(',')
+          labels_string_parts.each do |part|
+            key, value = part.split('=')
+            next if value.nil?
+            return value if key == match
+          end
+          'none'
+        end
+      end
+  
+      def update_user_metrics(operation, vmname)
+        tracer.in_span("Vmpooler::API::V2.#{__method__}") do |span|
+          begin
+            backend.multi
+            backend.hget("vmpooler__vm__#{vmname}", 'tag:jenkins_build_url')
+            backend.hget("vmpooler__vm__#{vmname}", 'token:user')
+            backend.hget("vmpooler__vm__#{vmname}", 'template')
+            jenkins_build_url, user, poolname = backend.exec
+            poolname = poolname.gsub('.', '_')
+  
+            if user
+              user = user.gsub('.', '_')
+            else
+              user = 'unauthenticated'
+            end
```

Comment:
Idiomatically, this would be better expressed as:

```ruby
user = if user
  user.gsub('.', '_')
else
  'unathenticated'
end
```

Also reusing `user` should be avoided. I would likely do something like:

```ruby
jenkins_build_url, default_user, poolname = backend.exec
user = if default_user
             default_user.gsub('.', '_')
           else
             'unauthenticated'
           end
```